lib/aws/rails/sesv2_mailer.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-sesv2'
+
+module Aws
+  module Rails
+    # Provides a delivery method for ActionMailer that uses Amazon Simple Email
+    # Service V2.
+    #
+    # Once you have an SESv2 delivery method you can configure Rails to
+    # use this for ActionMailer in your environment configuration
+    # (e.g. RAILS_ROOT/config/environments/production.rb)
+    #
+    #     config.action_mailer.delivery_method = :sesv2
+    #
+    # Uses the AWS SDK for Ruby's credential provider chain when creating an SESV2
+    # client instance.
+    class Sesv2Mailer
+      # @param [Hash] options Passes along initialization options to
+      #   [Aws::SESV2::Client.new](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SESV2/Client.html#initialize-instance_method).
+      def initialize(options = {})
+        @client = SESV2::Client.new(options)
+      end
+
+      # Rails expects this method to exist, and to handle a Mail::Message object
+      # correctly. Called during mail delivery.
+      def deliver!(message)
+        send_opts = {}
+        send_opts[:content] = {}
+        send_opts[:content][:raw] = { data: message.to_s }
+
+        # Check for Rails >=6 compatibility
+        send_opts[:from_email_address] = if message.respond_to?(:from_address)
+                                           message.from_address&.to_s
+                                         else
+                                           Array.wrap(message.from).first&.to_s
+                                         end
+
+        send_opts[:destination] = {}
+        send_opts[:destination][:to_addresses] = [*message.to]
+        send_opts[:destination][:cc_addresses] = [*message.cc]
+        send_opts[:destination][:bcc_addresses] = [*message.bcc]
```

Comment:
`Mail::Message#to`, `#cc` and `#bcc` reflects the `To`, `Cc` and `Bcc` mail header fields, but the mail should not necessarily be sent to the addresses specified in the header. You should use `Mail::Message#smtp_envelope_to` instead. See #94.

Reply:
Thanks @c960657. Any suggestions for this? According to the [AWS docs](https://docs.aws.amazon.com/ses/latest/APIReference-V2/API_Destination.html), we need to provide To/Cc/Bcc separately, but the smtp_envelope_to is a combination of all those fields right? So I'm not entirely sure how to use smpt_envelope_to here.

![image](https://user-images.githubusercontent.com/33270615/206944608-15776fd1-35fc-4a8c-93e7-d66955b1d795.png)

Reply:
It is unclear from the docs what these fields are used for, so it requires some experimenting. I would try just sticking all `smtp_envelope_to` addresses in the Bcc field and leave the others empty.

Reply:
I'll need to double check both APIs too. Thanks for bringing this up.